### PR TITLE
Use economic parameters in the objection function

### LINKED ISSUE #1050: Update objective function to use economic parameters

### PR DESCRIPTION
**Pull request by gnawin**
(description of changes)

Fixes #1050 

## Checklist before merging
- [x] Documentation is up-to-date
- [x] Unit tests have been added/updated accordingly
- [x] Code has been formatted according to SpineOpt's style
- [x] Unit tests pass

### CODE REVIEW

File: src/objective/connection_flow_costs.jl [hunk 1]
```diff
@@ -29,6 +29,9 @@ function connection_flow_costs(m::Model, t_range)
         sum(
             connection_flow[conn, n, d, s, t]
             * duration(t)
+            * (use_economic_representation(model=m.ext[:spineopt].instance) ?
+               connection_discounted_duration[(connection=conn, stochastic_scenario=s, t=t)] : 1
+            )
```
[the remark this answers is not in the document]
Comment (manuelma) [reply src/objective/connection_flow_costs.jl:34]: @gnawin @datejada

[the remark this answers is not in the document]
Comment (gnawin) [reply src/objective/connection_flow_costs.jl:34]: thank you for your comment @manuelma we will work on this

[the remark this answers is not in the document]
Comment (manuelma) [reply src/objective/connection_flow_costs.jl:34]: What is the connection_discounted_duration value you are using?

[the remark this answers is not in the document]
Comment (gnawin) [reply src/objective/connection_flow_costs.jl:34]: In this case, I'm testing another parameter `unit_conversion_to_discounted_annuities` but it throws the same error. 

`SpineOpt.unit_conversion_to_discounted_annuities(unit=unit(Symbol("power_plant")))` prints `SpineInterface.Map{Any, SpineInterface.TimeSeries{Any}}(Any[realization], SpineInterface.TimeSeries{Any}[TimeSeries[2000-01-01T00:00:00=>0.6139132535407585, 2005-01-01T00:00:00=>0.4810170980909696](ignore_year=false, repeat=false)])`, so for `realization` and e.g., `2005-01-01T00:00:00`, the parameter value is 0.4810170980909696. 

FYI, I'm using the below assignment (the latest PR #1085 added the last argument).
```
@eval begin
        $(param_name) = $(Parameter(param_name, [obj_cls]; mod = @__MODULE__))
end
```
Is there anything I should do specifically for this to work in my parameter generation?

[the remark this answers is not in the document]
Comment (gnawin) [reply src/objective/connection_flow_costs.jl:34]: Hi @manuelma, maybe you missed this, but do you have any idea how to continue this? Or if you need more information, just let me know. Thanks!

[the remark this answers is not in the document]
Comment (gnawin) [reply src/objective/connection_flow_costs.jl:34]: Thanks @manuelma. Good to know that maybe it's about SpineInterface. Yes, I'm fine with this. If you are also fine, let's keep it this way.

**Pull request merged**